node_modules/oae-authentication/lib/api.js

```diff
@@ -677,6 +679,113 @@ var getUserIdFromLoginId = module.exports.getUserIdFromLoginId = function(tenant
     _getUserIdFromLoginId(loginId, callback);
 };
 
+
+/**
+ * Get the secret for an existing user's resetpassword request
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {String}     tenantAlias     The tenant of the login id
+ * @param  {String}     userName        The user's own unique username 
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @param  {Object}     callback.secret The secret token that will be returned
+ */
+var getResetPasswordSecret =module.exports.getResetPasswordSecret = function(ctx, tenantAlias, username, callback){
+	
+    // Default the tenant alias to the current tenant's alias
+    tenantAlias = tenantAlias || ctx.tenant().alias;
+
+    // Create the loginid object first, so it can be passed into the validation
+    var loginId = new LoginId(tenantAlias, AuthenticationConstants.providers.LOCAL, username);
+    
+    // Check if we can get an existing userID using the userName the user provided
+    _getUserIdFromLoginId(loginId, function(err, userId) {
+        if (err) {
+            return callback({'code': 404, 'msg': 'No user could be found with the provided username'});
+        }
+
+        // Get user's object as we will try to send the user an email with token
+        PrincipalsDAO.getPrincipal(userId, function(err, user) {
+            if (err) {
+                return callback(err);
+            } else if (!user.email) {
+                log().warn({'userId': userId}, 'Used asked for password reset but has no email address');
+                return callback({'code': 400, 'msg': 'This user has no email address, you will have to ask an admin to reset the password'});
+            }
+
+            // Generate a secret token
+            var secret = crypto.randomBytes(16).toString('hex');
+
+            // The secret is stored for 24 hours
+            Cassandra.runQuery('UPDATE "AuthenticationLoginId" USING TTL 86400 SET "secret" = ? WHERE "loginId" = ?', [secret, _flattenLoginId(loginId)], function(err) {
```

Comment:
Thanks. I've created a file called addSecretColumnToAuthenticationLoginId.cql to keep the string for adding column `secret` into column family  `AuthenticationLoginId `. Would it be fine?